crates/meroctl/src/cli/config.rs

```diff
@@ -0,0 +1,185 @@
+
+            for item in listen_array.iter_mut() {
+                let addr: Multiaddr = item.as_str().unwrap().parse().unwrap();
+                let mut new_addr = Multiaddr::empty();
+
+                for protocol in addr.iter() {
+                    match protocol {
+                        Protocol::Ip4(_) if ipv4_host.is_some() => {
+                            new_addr.push(Protocol::Ip4(ipv4_host.unwrap()));
+                        }
+                        Protocol::Ip6(_) if ipv6_host.is_some() => {
+                            new_addr.push(Protocol::Ip6(ipv6_host.unwrap()));
+                        }
+                        Protocol::Tcp(_) | Protocol::Udp(_) if self.swarm_port.is_some() => {
+                            let new_port = self.swarm_port.unwrap();
+                            new_addr.push(match protocol {
+                                Protocol::Tcp(_) => Protocol::Tcp(new_port),
+                                Protocol::Udp(_) => Protocol::Udp(new_port),
+                                _ => unreachable!(),
+                            });
+                        }
+                        _ => new_addr.push(protocol),
+                    }
+                }
+
+                *item = Value::from(new_addr.to_string());
+            }
+        }
+
+        // Update server listen addresses
+        if !self.server_host.is_empty() || self.server_port.is_some() {
+            let server_table = doc["server"].as_table_mut().unwrap();
+            let listen_array = server_table["listen"].as_array_mut().unwrap();
+
+            for item in listen_array.iter_mut() {
+                let addr: Multiaddr = item.as_str().unwrap().parse().unwrap();
+                let mut new_addr = Multiaddr::empty();
+
+                for protocol in addr.iter() {
+                    match protocol {
+                        Protocol::Ip4(_) if ipv4_host.is_some() => {
+                            new_addr.push(Protocol::Ip4(ipv4_host.unwrap()));
+                        }
+                        Protocol::Ip6(_) if ipv6_host.is_some() => {
+                            new_addr.push(Protocol::Ip6(ipv6_host.unwrap()));
+                        }
+                        Protocol::Tcp(_) if self.server_port.is_some() => {
+                            new_addr.push(Protocol::Tcp(self.server_port.unwrap()));
+                        }
+                        _ => new_addr.push(protocol),
+                    }
+                }
+
+                *item = Value::from(new_addr.to_string());
+            }
+        }
+
+        // Update boot nodes if provided
+        if !self.boot_nodes.is_empty() {
+            let bootstrap_table = doc["bootstrap"].as_table_mut().unwrap();
+            let list_array = bootstrap_table["nodes"].as_array_mut().unwrap();
+            list_array.clear();
+            for node in self.boot_nodes.iter() {
+                list_array.push(node.to_string());
+            }
```

Comment:
like here.. what if I want to keep the existing boot nodes, but add a new one?

Reply:
I agree, my logic was:
- lets say the user initialized with calimero-dev boot nodes
- now he wants to switch to ipfs boot nodes

Should I just append ipfs boot nodes to already existing ones from calimero-dev?
To me it made sense that it overwrites the existing config.
Should I add an append flag or something similar?
Not sure about the cleanest way to handle the situation.

Reply:
> Should I add an append flag or something similar?

that just further complicates it, let's just give the user a flexible interface to mutate whatever data they want inside the config